### PR TITLE
Arcticus: update E4S-21.11 package list; add E4S-22.02, 22.05, 22.08 package lists and spack.yaml

### PR DESCRIPTION
**Pull request by koysean**

### CODE REVIEW

File: ANL/JLSE/Arcticus/E4S-22.05/e4s-22.05.txt [hunk 1]
```diff
@@ -0,0 +1,341 @@
+adios,1.13.1,oneapi@<version>,+blosc~bzip2~fortran~hdf5~infiniband+lz4+mpi~netcdf+shared+sz~szip+zfp+zlib patches=8ae17f6,aea47e5,d24b79b staging=none,linux-opensuse_leap15-icelake,l4xbjwj5mmnttkqldryz3vtl42a7xzyr
```
Comment (shahzebsiddiqui) [inline ANL/JLSE/Arcticus/E4S-22.05/e4s-22.05.txt:1]: was this a typo you have ``@<version>`` or intentional. I dont see any oneapi version in the output. Can you check the `spack find` command you ran.

File: ANL/JLSE/Arcticus/E4S-22.08/e4s-22.08.txt [hunk 2]
```diff
@@ -0,0 +1,329 @@
+adios,1.13.1,oneapi@<version>,+blosc~bzip2~fortran~hdf5~infiniband+lz4+mpi~netcdf+shared+sz~szip+zfp+zlib patches=8ae17f6,aea47e5,d24b79b staging=none,linux-opensuse_leap15-icelake,wccgwdwjo7bbw4lith4bsyb7lc7jag64
```
Comment (shahzebsiddiqui) [inline ANL/JLSE/Arcticus/E4S-22.08/e4s-22.08.txt:1]: same here you have `@<version>`

File: ANL/JLSE/Arcticus/E4S-22.08/spack.yaml [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    unify: false
+    reuse: false
+  compilers:
+  - compiler:
+      spec: oneapi@<version>
+      paths:
+        cc: /PATH/TO/icx
+        cxx: /PATH/TO/icpx
+        f77: /PATH/TO/ifx
+        fc: /PATH/TO/ifx
```
Comment (shahzebsiddiqui) [inline ANL/JLSE/Arcticus/E4S-22.08/spack.yaml:13]: was this intentional i see you have masked out these values? in your original spack configuration did you do `/path/to/icx` or you just did `cc: icx` i think you could probably get away with doing this

File: ANL/JLSE/Arcticus/E4S-22.08/spack.yaml [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+      flags: {}
+      operating_system: opensuse_leap15
+      target: x86_64
+      modules:
+        - ONEAPI-MODULES
```
Comment (shahzebsiddiqui) [inline ANL/JLSE/Arcticus/E4S-22.08/spack.yaml:18]: same here did you intentionally put `ONEAPI-MODULES` instead of the actual module name

**Comment by koysean** [comment]
Yes, all of the above is intentional - version numbers are protected under CNDA. I will put a comment at the top of the spack.yaml to explain this.

**Pull request merged**